core/meta/src/TEnum.cxx

```diff
@@ -195,9 +195,7 @@ TEnum *TEnum::GetEnum(const char *enumName, ESearchAction sa)
          return en;
       }
 
-      // Lock need for gROOT->GetListOfClasses() and the later update/modification to
-      // the autoparsing state.
-      R__LOCKGUARD(gInterpreterMutex);
+
```

Comment:
The subsequent `if` block seems to need at least a read lock, e.g. for `tClassScope->Property() & kIsNamespace` etc?

Reply:
Actually the lock is still held when executing this code and the subsequent code.  This portion of the code is a lambda function (assigned to variable `searchEnum`) which is only called when the read lock is held.